src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/Query/GeographicLevelFacetsParser.cs

```diff
@@ -0,0 +1,144 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Validators.ErrorDetails;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Requests;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.DuckDb;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model.Parquet.Tables;
+using InterpolatedSql;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Services.Query;
+
+internal class GeographicLevelFacetsParser(
+    QueryState queryState,
+    HashSet<GeographicLevel> allowedGeographicLevels)
+    : IFacetsParser
+{
+    public IInterpolatedSql Parse(DataSetQueryCriteriaFacets facets, string path)
+    {
+        var fragments = new List<IInterpolatedSql>();
+
+        var parsedEq = facets.GeographicLevels?.ParsedEq;
+
+        if (parsedEq is not null)
+        {
+            fragments.Add(
+                EqFragment(
+                    geographicLevel: parsedEq.Value,
+                    path: "geographicLevels.eq"
```

Comment:
Yep, we need this as it concatenates path segments together. We need the `path` as this is provided with any errors or warnings and directs the user to the part of the request that was causing the problem.

Reply:
Sorry, I think you misinterpreted what I was asking.

I was suggesting changing this line to `path: QueryUtils.Path(path, "geographicLevels.eq")` rather than the current `path: "geographicLevels.eq"`

Reply:
currently the `path` parameter passed into the `Parse` method isn't being used

Reply:
Ah sorry my bad, didn't read your comment correctly! I've added this in now, nice catch!